.github/workflows/post-dependabot.yml

```
@@ -22,12 +22,36 @@ jobs:
         with:
           go-version-file: .go-version
 
+      - name: set up git user
+        run: |
+          git config --global user.name 'dependabot[bot]'
+          git config --global user.email 'dependabot[bot]@users.noreply.github.com'
+          git remote set-url origin https://x-access-token:${{ secrets.GITHUB_TOKEN }}@github.com/${{ github.repository }}
```

IIUC, you can use:
- https://github.com/elastic/oblt-actions/tree/main/git/setup

Then it will configure the environment variable `GITHUB_TOKEN` and the git config settings

```suggestion
      - name: set up git user
        uses: elastic/oblt-actions/git/setup@v1
        with:
          username: 'dependabot[bot]'
          email: 'dependabot[bot]@users.noreply.github.com'
```